derive/src/lib.rs

```diff
@@ -3,31 +3,64 @@ extern crate proc_macro;
 use proc_macro::TokenStream;
 use proc_macro2::TokenStream as TokenStream2;
 use quote::quote;
-use syn::{Data, DeriveInput, Fields};
+use syn::{Data, DeriveInput, Field, Fields};
 
-#[proc_macro_derive(FromArgs)]
+#[proc_macro_derive(FromArgs, attributes(positional, keyword))]
 pub fn derive_from_args(input: TokenStream) -> TokenStream {
     let ast: DeriveInput = syn::parse(input).unwrap();
 
     let gen = impl_from_args(&ast);
     gen.to_string().parse().unwrap()
 }
 
+enum ArgType {
```

Comment:
I think this should be called `ArgKind` (since `Type` has a different meaning) and its variants (and the corresponding attributes) named according to the way Python does: https://docs.python.org/3/library/inspect.html#inspect.Parameter.kind (probably a good idea to include this link as well)